src/test/crux-messenger/test-secure-cruxid-messenger-prod-pubsubClientFactory.ts

```diff
@@ -0,0 +1,60 @@
+import * as chai from "chai";
+import sinon from "sinon";
+import chaiAsPromised from "chai-as-promised";
+import 'mocha';
+import {SecureCruxIdMessenger, CertificateManager} from "../../core/domain-services";
+import {BasicKeyManager, CruxNetPubSubClientFactory} from "../../infrastructure/implementations";
+import {InMemoryCruxUserRepository, MockUserStore, patchMissingDependencies} from "../test-utils";
+import {getMockUserBar123CSTestWallet, getMockUserFoo123CSTestWallet} from "./utils";
+
+patchMissingDependencies();
+
+chai.use(chaiAsPromised);
+chai.should();
+const expect = require('chai').expect;
+
+
+describe('Test Secure Crux Messenger - PROD', function() {
+    beforeEach(async function() {
+        const HOST = "127.0.0.1";
+        const PORT = 1883;
+        const userStore = new MockUserStore();
+        const user1Data = getMockUserFoo123CSTestWallet();
+        const user2Data = getMockUserBar123CSTestWallet();
+        this.user1Data = user1Data;
+        this.user2Data = user2Data;
+        userStore.store(user1Data.cruxUser);
+        userStore.store(user2Data.cruxUser);
+        this.inmemUserRepo = new InMemoryCruxUserRepository(userStore);
+        this.pubsubClientFactory = new CruxNetPubSubClientFactory({
+            defaultLinkServer: {
+                host: HOST,
+                port: PORT,
+            }
+        });
+    });
+
+    it('Basic Send Receive Test', async function() {
+        const testmsg = 'HelloWorld';
+        return new Promise(async (resolve, reject) => {
+            const user1Messenger = new SecureCruxIdMessenger(this.inmemUserRepo, this.pubsubClientFactory, {
+                cruxId: this.user1Data.cruxUser.cruxID,
+                keyManager: new BasicKeyManager(this.user1Data.pvtKey)
+            });
+
+            const user2Messenger = new SecureCruxIdMessenger(this.inmemUserRepo, this.pubsubClientFactory, {
+                cruxId: this.user2Data.cruxUser.cruxID,
+                keyManager: new BasicKeyManager(this.user2Data.pvtKey)
+            });
+            user2Messenger.listen((msg) => {
+                resolve(msg)
+            },(err) => {
+                reject(err)
+            });
+            await user1Messenger.send(testmsg, this.user2Data.cruxUser.cruxID);
+        }).then(msg => 
+            expect(msg).equals(testmsg)
```

Comment:
put this expectation before resolving promise on line 50. no need for this `then` statement

Reply:
did it for code symmetry (as in reject scenario) shall I still change it.

Reply:
@quixote911

Reply:
updated this